service/controller/resource/app/desired.go

```diff
@@ -415,9 +419,9 @@ func (r *Resource) getCatalogIndex(ctx context.Context, catalogName string) ([]b
 	client := &http.Client{}
 
 	var err error
-	var catalog *g8sv1alpha1.AppCatalog
+	catalog := &g8sv1alpha1.AppCatalog{}
 	{
-		catalog, err = r.g8sClient.ApplicationV1alpha1().AppCatalogs().Get(ctx, catalogName, metav1.GetOptions{})
+		err = r.ctrlClient.Get(ctx, types.NamespacedName{Name: catalogName}, catalog)
```

Comment:
maybe let's make namespace explicit here? I guess it's "default"

Reply:
its not namespaced, it's not needed

Reply:
ok!